File: unit1/calc.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>

float find_mod(float, float);

int main(int argc, string argv[])
{
    if (argc != 4)
    {
        printf("Usage: ./calc <number> <op> <number>\n");
        return 1;
    }

    float x = atof(argv[1]);
    // float x = (float) strtod(argv[1], NULL);

    float y = atof(argv[3]);
    // float y = (float) strtod(argv[1], NULL);

    char op = argv[2][0];

    if (op == '+')
    {
        printf("%f\n", x + y);
    }
    else if (op == '-')
    {
        printf("%f\n", x - y);
    }
    else if (op == 'x')
    {
        printf("%f\n", x * y);
    }
    else if (op == '/')
    {
        printf("%f\n", x / y);
    }
    else if (op == '%')
    {
        printf("%f\n", find_mod(x, y));
    }
    else
    {
        printf("Invalid operator.\n");
        return 1;
    }
}

float find_mod(float a, float b)
{
    int q = (int) (a / b);
    float rem = a - (b * q);
    return rem;
}

File: unit3/calc.py
from sys import argv, exit

def find_mod(a, b):
    q = int(a / b)
    rem = a - (b * q)
    return rem

def main():
    if len(argv) != 4:
        print("Usage: python calc.py <number> <op> <number>")
        exit(1)
    
    x = float(argv[1])
    y = float(argv[3])
    
    op = argv[2]
    
    if op == "+":
        print(x + y)
    elif op == "-":
        print(x - y)
    elif op == "x":
        print(x * y)
    elif op == "/":
        print(x / y)
    elif op == "%":
        print(find_mod(x, y))
    else:
        print("Invalid operator")
        exit(1)

main()